src/cai-block.ads

```diff
@@ -1,56 +1,121 @@
 package Cai.Block
    with SPARK_Mode
 is
-   pragma Compile_Time_Error (Byte'Size /= 8,
-                              "Byte must have a size of 8 bit.");
+   pragma Compile_Time_Error (Byte'Size /= 8, "Byte must have a size of 8 bit.");
 
+   --  Number of Bytes
    type Byte_Length is range 0 .. 2 ** 63 - 1
       with Size => 64;
+   --  Block Id
    type Id is mod 2 ** 64
       with Size => 64;
+   --  Amount of blocks
    type Count is range 0 .. 2 ** 63 - 1
       with Size => 64;
+   --  Size of a block in bytes
    type Size is range 0 .. 2 ** 32 - 1
       with Size => 64;
 
+   --  * Operator for block count and size
+   --
+   --  @param Left   Block count
+   --  @param Right  Block size
+   --  @return       Byte size of all blocks
    function "*" (Left : Count; Right : Size) return Byte_Length is
       (Byte_Length (Left * Count (Right)));
+   --  * Operator for block count and size
+   --
+   --  @param Left   Block size
+   --  @param Right  Block count
+   --  @return       Byte size of all blocks
    function "*" (Left : Size; Right : Count) return Byte_Length is
       (Right * Left);
+   --  * Operator for block count and size
+   --
+   --  @param Left   Block size
+   --  @param Right  Block count
+   --  @return       Buffer length
    function "*" (Left : Count; Right : Size) return Buffer_Index is
       (Buffer_Index (Left * Count (Right)));
+   --  * Operator for block count and size
+   --
+   --  @param Left   Block size
+   --  @param Right  Block count
+   --  @return       Buffer length
    function "*" (Left : Size; Right : Count) return Buffer_Index is
       (Right * Left);
+   --  + Operator for block Id and count
+   --
+   --  @param Left   Block Id
+   --  @param Right  Block count
+   --  @return       Block Id that has offset Right from Left
    function "+" (Left : Id; Right : Count) return Id is
       (Left + Id (Right));
+   --  - Operator for block Id and count
+   --
+   --  @param Left   Block Id
+   --  @param Right  Block count
+   --  @return       Block Id that has offset -Right from Left
    function "-" (Left : Id; Right : Count) return Id is
       (Left - Id (Right));
+   --  - Operator for Block Ids
+   --
+   --  @param Left   Block Id
+   --  @param Right  Block Id
+   --  @return       Offset between Left and Right
    function "-" (Left : Id; Right : Id) return Count is
       (Count (Left) - Count (Right)) with
       Pre => Left >= Right;
 
+   --  Type of a block request
+   --
+   --  @value None   Invalid request
+   --  @value Read   Read request
+   --  @value Write  Write request
+   --  @value Sync   Sync request
+   --  @value Trim   Trim request
    type Request_Kind is (None, Read, Write, Sync, Trim);
+   --  Status of a block request
+   --
+   --  @value Raw           Newly created request
+   --  @value Ok            Successfully handled request
+   --  @value Error         Failed request
+   --  @value Acknowledged  deprecated
```

Comment:
Why can't we remove `Acknowledged`?

Reply:
The deprecated is already deprecated ;) The `Acknowledged` status is required for the server to know if an acknowledgement has been done on the request by the platform. If the platform is currently unavailable to acknowledge a request this is the way to detect this for the server. Although I have to say that this interface is not fully conformant with our approach to check first and then do an always succeeding call. We should check how this can be improved and how it would fit into Genode (I'm afraid it won't without dirty hacks).